File: src/main/java/com/github/kpavlov/jreactive8583/server/ServerConfiguration.java
package com.github.kpavlov.jreactive8583.server;

import com.github.kpavlov.jreactive8583.ConnectorConfiguration;

public class ServerConfiguration extends ConnectorConfiguration {

    public ServerConfiguration(final Builder builder) {
        super(builder);
    }

    public static Builder newBuilder() {
        return new Builder();
    }

    public static ServerConfiguration getDefault() {
        return newBuilder().build();
    }

    public static class Builder extends ConnectorConfiguration.Builder<Builder> {
        public ServerConfiguration build() {
            return new ServerConfiguration(this);
        }
    }
}

File: src/main/kotlin/com/github/kpavlov/jreactive8583/server/ServerConfiguration.kt
@file:JvmName("ServerConfiguration")

package com.github.kpavlov.jreactive8583.server

import com.github.kpavlov.jreactive8583.ConnectorConfiguration

@SuppressWarnings("WeakerAccess")
public class ServerConfiguration(
    builder: Builder
) : ConnectorConfiguration(builder) {

    public companion object {
        @JvmStatic
        public fun newBuilder(): Builder = Builder()

        @Suppress("unused")
        @JvmStatic
        public fun getDefault(): ServerConfiguration = newBuilder().build()
    }

    public class Builder : ConnectorConfiguration.Builder<Builder>() {
        public fun build(): ServerConfiguration = ServerConfiguration(this)
    }
}
